decision/blockstoremanager.go

```diff
@@ -100,19 +104,23 @@ func (bsm *blockstoreManager) getBlocks(ctx context.Context, ks []cid.Cid) map[c
 			lk.Unlock()
 		}
 	})
-
-	return res
 }
 
-func (bsm *blockstoreManager) jobPerKey(ctx context.Context, ks []cid.Cid, jobFn func(c cid.Cid)) {
+func (bsm *blockstoreManager) jobPerKey(ctx context.Context, ks []cid.Cid, jobFn func(c cid.Cid)) error {
+	var err error
 	wg := sync.WaitGroup{}
 	for _, k := range ks {
 		c := k
 		wg.Add(1)
-		bsm.addJob(ctx, func() {
+		err = bsm.addJob(ctx, func() {
 			jobFn(c)
 			wg.Done()
 		})
+		if err != nil {
+			wg.Done()
+			break
```

Comment:
We can probably just return the error directly here instead of breaking out of the loop right

Reply:
I'd like to wait for the jobs to complete. Otherwise, the closer could still be writing to shared state.

Reply:
Makes sense 👍